src/store/item.js

```diff
@@ -1,4 +1,13 @@
 import { writable } from 'svelte/store';
+import colorOptions from '@/constants/item/colors';
+
+export const colors = writable(colorOptions);
+export const customColors = writable({});
+export function addNewColor(id, color) {
+  customColors.update(x => {
```

Comment:
Check if this color already exists first before adding it

Reply:
Right now it doesn't matter, because we don't have real color-picker and all the colors are not well-defined. The implemented logic is a "close interpretation of real one"